lib/web/apiserver.go

```diff
@@ -5387,6 +5364,7 @@ func ParseSSORequestParams(r *http.Request) (*SSORequestParams, error) {
 	}
 
 	query := r.URL.Query()
+	username := query.Get("username")
```

Comment:
We take this user-provided value, and use it in a gRPC call to get the IDP's authorization URL. gRPC messages have a limit to how large they can be, so if a user puts in a really long username, they might see a weird gRPC failure.

If the URL param is greater than say 1024 bytes, then we can just drop it here instead of passing it through. That will prevent weird errors from occuring.